File: mgsr-web/src/lib/playerTaskTemplates.ts
/**
 * Predefined task templates for player-related tasks.
 * Data is fetched from Firestore remote config (with hardcoded fallbacks).
 */
import { appConfig, type TaskTemplate } from '@/lib/appConfig';

export type PlayerTaskTemplate = TaskTemplate;

export const PLAYER_TASK_TEMPLATES: PlayerTaskTemplate[] = new Proxy([] as PlayerTaskTemplate[], {
  get(target, prop) {
    const templates = appConfig.taskTemplates.templates;
    if (prop === 'length') return templates.length;
    if (prop === Symbol.iterator) return templates[Symbol.iterator].bind(templates);
    if (typeof prop === 'string' && !isNaN(Number(prop))) return templates[Number(prop)];
    // Proxy array methods like map, forEach, etc.
    const val = (templates as any)[prop];
    return typeof val === 'function' ? val.bind(templates) : val;
  },
});

export function getTemplateTitle(template: PlayerTaskTemplate, lang: 'en' | 'he', month?: number, isWomen?: boolean): string {
  const monthsEN = appConfig.taskTemplates.monthsEN;
  const monthsHE = appConfig.taskTemplates.monthsHE;
  const useWomen = isWomen && (lang === 'he' ? template.titleHeWomen : template.titleEnWomen);
  const title = lang === 'he'
    ? (useWomen ? template.titleHeWomen! : template.titleHe)
    : (useWomen ? template.titleEnWomen! : template.titleEn);
  if (template.hasMonthPlaceholder && month != null) {
    const monthName = lang === 'he' ? monthsHE[month] : monthsEN[month];
    return title.replace('{month}', monthName);
  }
  if (template.hasMonthPlaceholder) {
    return title.replace('{month}', 'X');
  }
  return title;
}

File: app/src/main/java/com/liordahan/mgsrteam/features/home/tasks/PlayerTaskTemplates.kt
package com.liordahan.mgsrteam.features.home.tasks

import com.liordahan.mgsrteam.config.AppConfigManager

/**
 * Predefined task templates for player-related tasks.
 * Data is fetched from Firestore remote config (with hardcoded fallbacks).
 */
data class PlayerTaskTemplate(
    val id: String,
    val titleEn: String,
    val titleHe: String,
    val hasMonthPlaceholder: Boolean = false
)

val PLAYER_TASK_TEMPLATES: List<PlayerTaskTemplate>
    get() = AppConfigManager.taskTemplates.templates.map { t ->
        PlayerTaskTemplate(t.id, t.titleEn, t.titleHe, t.hasMonthPlaceholder)
    }

private val MONTHS_EN: List<String>
    get() = AppConfigManager.taskTemplates.monthsEN

private val MONTHS_HE: List<String>
    get() = AppConfigManager.taskTemplates.monthsHE

fun getTemplateTitle(template: PlayerTaskTemplate, isHebrew: Boolean, month: Int? = null): String {
    val title = if (isHebrew) template.titleHe else template.titleEn
    return if (template.hasMonthPlaceholder && month != null && month in 0..11) {
        val monthName = if (isHebrew) MONTHS_HE[month] else MONTHS_EN[month]
        title.replace("{month}", monthName)
    } else if (template.hasMonthPlaceholder) {
        title.replace("{month}", "X")
    } else {
        title
    }
}
